File: DMPreview/HD-DM-Android-SDK-Release/DMPreview/app/src/main/java/com/esp/uvc/usbcamera/SettingsMainActivity.java
package com.esp.uvc.usbcamera;

import android.os.Bundle;
import android.preference.PreferenceFragment;
import androidx.appcompat.app.AppCompatActivity;

import com.esp.uvc.R;


public class SettingsMainActivity extends AppCompatActivity {
    private static final boolean DEBUG = true;
    private static final String TAG = "SettingsMianActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         getFragmentManager().beginTransaction()
             .replace(android.R.id.content, new SettingsFragment())
             .commit();
    }

    @Override
    public void onResume() {
        super.onResume();

    }

    public static class SettingsFragment extends PreferenceFragment {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            addPreferencesFromResource(R.xml.settings_main);
        }
    }

}

File: DMPreview/HD-DM-Android-SDK-Release/DMPreview/app/src/main/java/com/esp/uvc/main/settings/SettingsMainActivity.kt
package com.esp.uvc.main.settings

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import com.esp.uvc.BaseFragment
import com.esp.uvc.R
import com.esp.uvc.main.settings.module_sync.ModuleSyncActivity
import com.esp.uvc.main.settings.sensor.SensorSettingsActivity
import kotlinx.android.synthetic.main.activity_settings_main.*

class SettingsMainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        BaseFragment.applyScreenRotation(this)
        super.onCreate(savedInstanceState)
        title = getString(R.string.settings_main_title)
        setContentView(R.layout.activity_settings_main)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        initViews()
    }

    private fun initViews() {
        layout_preview_image.setOnClickListener {
            PreviewImageActivity.startInstance(this)
        }
        layout_sensor_settings.setOnClickListener {
            SensorSettingsActivity.startInstance(this)
        }
        layout_ply_viewer.setOnClickListener {
            PlyViewerActivity.startInstance(this)
        }
        layout_firmware_version.setOnClickListener {
            showFirmwareVersion()
        }
        layout_firmware_register.setOnClickListener {
            FirmwareRegisterActivity.startInstance(this)
        }
        layout_firmware_table.setOnClickListener {
            FirmwareTableActivity.startInstance(this)
        }
        layout_software_version.setOnClickListener {
            showSoftwareVersion()
        }
        layout_module_sync.setOnClickListener {
            ModuleSyncActivity.startInstance(this)
        }
    }

    private fun showFirmwareVersion() {
        FirmwareVersionActivity.startInstance(this)
    }

    private fun showSoftwareVersion() {
        val fragment = SoftwareVersionDialogFragment()
        fragment.show(supportFragmentManager, "softwareVersion")
    }

    companion object {
        fun startInstance(context: Context) {
            context.startActivity(Intent(context, SettingsMainActivity::class.java))
        }
    }
}